x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_icon_property.tsx

```diff
@@ -33,12 +37,23 @@ export class DynamicIconProperty extends DynamicStyleProperty<IconDynamicOptions
 
   syncIconWithMb(symbolLayerId: string, mbMap: MbMap, iconPixelSize: number) {
     if (this._isIconDynamicConfigComplete()) {
-      mbMap.setLayoutProperty(
-        symbolLayerId,
-        'icon-image',
-        this._getMbIconImageExpression(iconPixelSize)
-      );
-      mbMap.setLayoutProperty(symbolLayerId, 'icon-anchor', this._getMbIconAnchorExpression());
+      const { customIconStops, iconPaletteId } = this._options;
+      const icons = [
+        ...(iconPaletteId ? getIconPalette(iconPaletteId) : []),
+        ...(customIconStops ? customIconStops.map(({ icon }) => icon) : []),
+      ];
+      Promise.all(
```

Comment:
Can you add logic to handle exceptions. For example, `loadMakiIconInMap` could throw an error if `getMakiSymbolSvg` gets an unknown id. I think the best way to handle the error is to surface a warning toast.